### DISCUSSION TITLE
Not able to deploy edge function

### DISCUSSION THREAD

**Discussion by CareTiger** [supabase/supabase#14177] [Questions]
I am trying to deploy multiple edge functions and one of them fails with the following error
`Bundling contact
Error: Error bundling function: exit status 1
file:///src/import_map.json
file:///src/index.ts
file:///_services/contactMeService.ts
file:///_shared/cors.ts
file:///_shared/types.ts
file:///types/db_types.ts
error: Uncaught (in promise) Error: NotFound: The system cannot find the path specified. (os error 3)
      const ret = new Error(getStringFromWasm0(arg0, arg1));
                  ^
    at __wbg_new_8d2af00bc1e329ee (https://deno.land/x/eszip@v0.35.0/eszip_wasm.generated.js:513:19)
    at <anonymous> (https://deno.land/x/eszip@v0.35.0/eszip_wasm_bg.wasm:1:1559899)
    at <anonymous> (https://deno.land/x/eszip@v0.35.0/eszip_wasm_bg.wasm:1:1398157)
    at <anonymous> (https://deno.land/x/eszip@v0.35.0/eszip_wasm_bg.wasm:1:1895031)
    at __wbg_adapter_40 (https://deno.land/x/eszip@v0.35.0/eszip_wasm.generated.js:229:6)
    at real (https://deno.land/x/eszip@v0.35.0/eszip_wasm.generated.js:213:14)`  the edge function works locally.

**Comment by cpannwitz** [comment]
Had the same problem, trying to import the local file with generated supabase DB types.
Did not find a solution yet.

**Comment by laktek** [comment]
@CareTiger Do you still experience this issue with the latest CLI (1.61.5+)

**Comment by CareTiger** [comment]
@laktek am on the road and dont have access to my computer. will get my India team to try it out when they come in and revert

**Comment by CareTiger** [comment]
@laktek  Nope. This is not working still.. Giving error in local

**Comment by CareTiger** [comment]
we have the same problem as this discord user - https://discord.com/channels/839993398554656828/1105999104984748102/1108982356922081290

**Comment by sampatsharma143** [comment]
I was able to solve this issue by downgrading my supabase cli version  from 1.62.3 to 1.61.0
I used scoop to install the supabase ( official doc) and there was no option to downgrade .
```
1# so i installed the latest version of supabase using scoop ,  
2# find the location of the supabase cli exe file - > it was ( C:\Users\91995\scoop\apps\supabase\current) for me.
3# downloaded the 1.61.0 version from the github release section  - https://github.com/supabase/cli/releases/tag/v1.61.0 
4# extracted the supabase exe and replaced it with latest version found in #2 ( C:\Users\91995\scoop\apps\supabase\current)
```

Hope it will help , i don't want someone to suffer like it did . peace!

**Reply by Lewitje** [reply]
This fixed it for me, thanks!

**Reply by sampatsharma143** [reply]
I am glad I was able to help.

**Reply by CareTiger** [reply]
@sampatsharma143 glad it worked for you but the speed at which these cli versions are being released and the constant changes lead me to believe that this is a temporary solution and wont work for us. thanks for sharing though.

**Reply by sampatsharma143** [reply]
Yeah but we can switch to a latest working version as soon as they release it

**Reply by sweatybridge** [reply]
I can reproduce this issue on windows. It will be addressed in latest release 1.62.4.

**Reply by sampatsharma143** [reply]
@sweatybridge  great news , thanks

**Reply by michael-sumner** [reply]
I was facing an issue with this too. Running the following fixed it:

```sh
npx supabase@1.61.0 functions deploy <function_name> --project-ref <your_project_id>
```

**Comment by kkukshtel** [comment]
I'm seeing this issue again on v1.110.1

```
error: Uncaught (in promise) Error: NotFound: No such file or directory (os error 2)
      const ret = new Error(getStringFromWasm0(arg0, arg1));
                  ^
    at __wbg_new_8d2af00bc1e329ee (https://deno.land/x/eszip@v0.35.0/eszip_wasm.generated.js:513:19)
    at <anonymous> (https://deno.land/x/eszip@v0.35.0/eszip_wasm_bg.wasm:1:1559899)
    at <anonymous> (https://deno.land/x/eszip@v0.35.0/eszip_wasm_bg.wasm:1:1398157)
    at <anonymous> (https://deno.land/x/eszip@v0.35.0/eszip_wasm_bg.wasm:1:1895031)
    at __wbg_adapter_40 (https://deno.land/x/eszip@v0.35.0/eszip_wasm.generated.js:229:6)
    at real (https://deno.land/x/eszip@v0.35.0/eszip_wasm.generated.js:213:14)
```

EDIT: Resolved this. My function file wasn't named ìndex.ts` and instead was called `server.ts`. Changing the file name fixed it.

**Comment by florian-mlr** [comment]
I'm facing this issue on v1.112.0 . My function file is `index.ts`

**Reply by Keegan-lee** [reply]
Same - Will post if I find the solution..

**Comment by Tombcat** [comment]
For me the problem was autogenerated types using supabase cli. I Had to manually download file from the dashboard https://supabase.com/dashboard/project/_/api?page=tables-intro

**Comment by ljhskyso** [comment]
same issue.

**Comment by code2fire00** [comment]
### solution that worked for me (cli version 1.115.4) :
Change the file encoding to UTF-8.


### Potential cause:
I've noticed that the generated file encoding format is utf-16 LE, that causes the error.
It works when the file is encoded in utf-8.

Its because PowerShell creates the generated file in utf-16, its not a supabase cli problem.

![image](https://github.com/supabase/supabase/assets/98121073/9d04dd75-33e8-4fd0-8419-6acbc0a74785)


---

FYI:
Switching versions didn't work for me

**Comment by zdzarski** [comment]
Usually, this is not working when your project structure is wrong. I was able to fix it verifying if my directory structure is correct (`_shared` was not in a functions directory).

https://supabase.com/docs/guides/functions/quickstart#organizing-your-edge-functions

**Reply by code2fire00** [reply]
_shared is a folder that contains shared typescript modules between my edge functions. But the edge functions do not support demo file system api, so we cannot put html in the _shared folder, just Es modules. Btw shared is just a name I gave you can name it whatever u want but needs to be prefixed with _

**Reply by zdzarski** [reply]
That's great explanation, thanks.

**Comment by codingiswhyicry** [comment]
Facing this same error looking to deploy an edge function, and realized I mistyped the name of the actual function itself. So, seems like this can be a very generic and open-ended error. 

Catch my error below, my function was named inpesction instead of inspection. 
--- 
Supabase CLI 1.115.5
Version 1.30.3 is already installed
Bundling finish_inpesction
Error bundling function: exit status 1                                                                                 
file:///Users/supabase/functions/import_map.json           
file:///Users/supabase/functions/finish_inpesction/index.ts
error: Uncaught (in promise) Error: NotFound: No such file or directory (os error 2)                                   
      const ret = new Error(getStringFromWasm0(arg0, arg1));

**Reply by HassanDev13** [reply]
Me to lol

**Comment by JoshuaPerk** [comment]
Same problem, until I realized I was importing a filed called `cors.ts` when I had actually named is `cors.js`

**Comment by nsnider7** [comment]
For those here now, make sure all your functions locally have a index.ts file in them. I had a couple random folders that were generated by cursor that were empty so I got this error

**Comment by conn-do** [comment]
I encountered this issue when deploying through the GitHub integration (CLI deployment worked fine).
It turned out to be caused by an unnecessary deno.lock file. Removing it fixed the problem.
